apps/web/app/(app)/stats/NewsletterModal.tsx

```diff
@@ -52,13 +71,23 @@ export function NewsletterModal(props: {
             </DialogHeader>
 
             <div className="flex space-x-2">
-              <Button size="sm" variant="outline">
+              <Button size="sm" variant="outline" disabled={unsubscribeLoading}>
                 <a
                   href={newsletter.lastUnsubscribeLink || undefined}
                   target="_blank"
                   rel="noreferrer"
+                  onClick={onUnsubscribe}
                 >
-                  Unsubscribe
+                  {unsubscribeLoading ? (
+                    // Button loading state
```

Comment:
really don't need this comment lol. pretty clear from the code that it's loading state 😆

Reply:
Haha done, removed the comments now